java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngine.java

```diff
@@ -31,19 +32,26 @@ public class DbzCdcEngine implements CdcEngine {
     private final long id;
 
     /** If config is not valid will throw exceptions */
-    public DbzCdcEngine(long id, Properties config, DebeziumEngine.CompletionCallback callback) {
+    public DbzCdcEngine(
+            long sourceId,
+            Properties config,
+            CountDownLatch startSignal,
+            DebeziumEngine.CompletionCallback completionCallback) {
```

Comment:
I saw that a [ConnectorCallback](https://github.com/debezium/debezium/blob/8c1c369449b17a938c5d726058051048c80661fb/debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java#L73) can be passed to the dbz engine. Have we investigated whether this callback can achieve what we need instead of relying on heartbeat?

Reply:
The debezium engine will call the callback just after it submits the connector task to a worker thread. But what we want is to align with the time point that connector has subscribed to the binlog. So this interface is not reliable to us.